x-pack/plugins/fleet/server/services/agents/upgrade_action_runner.ts

```diff
@@ -157,15 +157,13 @@ const getRollingUpgradeOptions = (startTime?: string, upgradeDurationSeconds?: n
         MINIMUM_EXECUTION_DURATION_SECONDS,
         upgradeDurationSeconds
       ),
-      expiration: NO_EXPIRATION,
```

Comment:
using default 1M expiration as discussed here https://github.com/elastic/kibana/issues/146018#issuecomment-1369727911

Reply:
@juliaElastic I think we probably still want to introduce a new end time property for fleet server as otherwise it will try to spread actions between start time and expiration.

Reply:
the previous expiration was [calculated](https://github.com/elastic/kibana/pull/148240/files#diff-3ee1b1cbc1b2fcebb09cef6838b5a06980ae475d5001c07ac7d0c2aced9bbfad) by `start time + rollout period`, I think Fleet Server could do the same calculation, so we wouldn't need an end time

Reply:
> the previous expiration was [calculated](https://github.com/elastic/kibana/pull/148240/files#diff-3ee1b1cbc1b2fcebb09cef6838b5a06980ae475d5001c07ac7d0c2aced9bbfad) as start time + rollout period, I think Fleet Server could do the same calculation, so we wouldn't need an end time

Yes we could do this too, but we will still have to make a change to pass the rollout period to fleet server in the `.fleet-actions` index.

Reply:
it is passed currently in the `minimum_execution_duration` field, I thought we can use the same

Reply:
If I remember correctly the `minimum_execution_duration` is another field that is a constant defining the minimum time the agent should have to execute the action 

it looks like something like this
<img width="1147" alt="Screen Shot 2023-01-03 at 10 07 31 AM" src="https://user-images.githubusercontent.com/1336873/210373235-093e659a-0e96-48da-a00a-990e58ebe7d3.png">

Reply:
Yes, I see that this was the logic in Fleet Server, however the [logic](https://github.com/elastic/kibana/pull/148240/files#diff-3ee1b1cbc1b2fcebb09cef6838b5a06980ae475d5001c07ac7d0c2aced9bbfad) in Fleet API used to set the `expiration=start time + min execution time`, and the min execution time comes from the UI `Maintenance window` field.
So I don't really understand how this worked, because `expiration - min execution time == start time`, so it sounds like the period was 0. I'll test this locally.

My thinking was that if we eliminate expiration from the calculation, we get `start_time : start_time+min execution time (rollout period)` range.

Reply:
Looks like the logic was setting expiration used upgradeDurationSeconds another variable pass through the API:
```
expiration: moment(startTime ?? now)
        .add(upgradeDurationSeconds, 'seconds')
        .toISOString(),
```

Reply:
@juliaElastic We should backport this fix to 8.6.0 as well.

Reply:
Isn't it too late? 8.6.0 is on Jan 10, and this contains a breaking change. I have 3 open prs in 3 projects, and trying to test this including the upgrade sceenarios.

Reply:
Sorry to clarify I was thinking that we should consider backporting a new, isolated fix to fix the math so that the <2 hour window works to a future 8.6.x patch release, if it's feasible. I agree we shouldn't backport this entire change to 8.6.

I also think that if it's non-trivial to do, we can just leave it as-is in 8.6 and rely on these changes landing in 8.7.0

Reply:
Raised a fix for 8.6 https://github.com/elastic/kibana/pull/148502